### ISSUE TITLE
Fuse.py may not work properly 

### ISSUE THREAD

**Issue by Doriandarko** [ml-explore/mlx-examples#425]
When running inference on a finished lora using the created adapter, example:

python lora.py --model "mlx-community/Nous-Hermes-2-Mixtral-8x7B-DPO-4bit" \
               --adapter-file "mixtral.npz" \
               --max-tokens 50 \
               --prompt "Question: ELI5 how to make great pizza./n Assistant:"
               
               
You seem to get the corrected finetuned answer, instead when running the inference on you lora fused model, you get the same exact answer as the original model

**Comment by mzbac** [comment]
There was a bug in mlx-lm that caused the lora not to apply during fuse. Would you try installing the latest mlx-lm to see if it fixes the issue?
```
pip install -U mlx-lm
```

**Comment by Doriandarko** [comment]
Ok let me update, fuse the model again and upload

**Comment by mzbac** [comment]
> Ok let me update, fuse the model again and upload

Sorry for the inconvenience, and I am thinking maybe we can find a better way to share the mlx model with adapters instead of using a fused model. because during the de-quantization process, there will be some performance impact caused by quantization rounding.

**Comment by awni** [comment]
We could easily upload the adapters to HF as well. In fact that's really all we'd need to upload as long as we track the base model..

That said it is faster for inference to use a pre-fused model.. so one may want to do that anyway.

**Comment by mzbac** [comment]
> We could easily upload the adapters to HF as well. In fact that's really all we'd need to upload as long as we track the base model..
> 
> That said it is faster for inference to use a pre-fused model.. so one may want to do that anyway.

Yeah, we should be able to support both. And mlx doesn't have many legacy issues, so including the adapters with the base models, sounds very good to me.
